my card is not visible in the html?

AnaCascante opened the pull request:
starting to work with Js of hell

Commit:
my card is not visible in the html?

AnaCascante committed:
working on it...

AnaCascante committed:
changes

AnaCascante committed:
doing some changes

AnaCascante committed:
js from hell!

Jean-Christoffer commented on html/feed/feed.js:
```diff
@@ -24,3 +24,60 @@ async function getPosts (headerOptions){
+
+
+async function createCards (){
+        const posts = await getPosts()
```
You are not seing anything because you call the function with no options, you call the function outside the createcards function with the correct parameters. You dont need to call the function two times. You want the data to use inside the createCards function so you only need to call it inside that function, and with the correct parameters, in this case options.

AnaCascante replied:
this code is removed

Jean-Christoffer commented:
Remember to tag me when you feel you are ready for a review :) also write descriptive summaries/comments so its easier to understand what changes has been done

AnaCascante committed:
working with html in feed

AnaCascante committed:
why are not the post in the page.

I think post should be filtrated in line 29- to get just post from noroff.. -- could it be an idea to make a tag when they register- tag: student and tag:teacher- so add <_tag=student&_tag=teacher> to the url

Jean-Christoffer commented on html/feed/feed.js:
```diff
@@ -1,5 +1,369 @@
+  try {
+      const response = await fetch (`${baseUrl}/social/posts?_author=true&_comments=true&_reactions=true`,headerOptions); 
+      const posts= await response.json();
+      console.log (posts); 
```
Remember to remove console.logs when you want to push to Main branch as debugging should have been completed and the app doesnt get overwhelemed with logs at the end :p

AnaCascante replied:
fixed

Jean-Christoffer reviewed (changes requested):
Well done, im reviewing from mobile but the code looks good. If you have tested it and the posts generates html you can squash and merge,  also remove the console.log

Jean-Christoffer commented on html/feed/feed.html:
```diff
@@ -5,37 +5,155 @@
+        <div class="collapse navbar-collapse" id="navbarNav">
+        <ul class="navbar-nav ml-auto">
+          <li class="nav-item">
+          <a class="nav-link" href="/html/feed.html">Nerdsnett</a>
```
does not navigate to the correct page

AnaCascante replied:
fixed

Jean-Christoffer commented on html/feed/feed.html:
```diff
@@ -5,37 +5,155 @@
+          <a class="nav-link" href="/html/feed.html">Nerdsnett</a>
+          </li>
+          <li class="nav-item">
+          <a class="nav-link" href="/html/profile.html">My profile</a>
```
does not navigate to the correct page

AnaCascante replied:
fixed

Jean-Christoffer commented on html/feed/feed.html:
```diff
@@ -5,37 +5,155 @@
 
-        <a class="navbar-brand">
+     
+        <a class="navbar-brand" href="/html/feed/feed.html">
```
this is correct : )

Jean-Christoffer commented on html/feed/feed.js:
```diff
@@ -1,5 +1,369 @@
+      commentsPanel.style.display = 'none'; 
+    } 
+
+  container.appendChild (card); 
```
you are appending the html to main inside a eventlistner for the comment button. please check your scoping when writing functions. the container which is the main tagg in this case, appends this card component when the function runs. if it only appends the the card when the comment button is pressed which doesnt exist yet because main has not yet appended the html, then it cant render :P move this out of the eventlistener please.

AnaCascante replied:
Thank you. I see the error. I moved the append to be in the function.

Jean-Christoffer replied:
its still not in the correct part, the card is defined in the forEach part of the function. meaning forEach post it creats a card element, we then need to append the Card element each time a card has been made. now you are appending one card that doesnt exist to the container element outside the forEach function. move it inside the forEach function and it should work.
![image](https://github.com/AnaCascante/2023-10-15_JS2_CA_AnaC_JeanC_Repo/assets/89041011/4eba9f18-0940-4a31-88a2-3e6b2c8063d7)

AnaCascante replied:
fixed

Jean-Christoffer reviewed (changes requested):
please check my comments : ) it will work if you make the changes i commented on

Jean-Christoffer commented on html/feed/feed.js:
```diff
@@ -1,5 +1,369 @@
+    colImage.className = 'col-md-auto py-2 pl-3';
+
+    const userImg = document.createElement ('div'); 
+    userImg.src = element.author.avatar ?? "../../img/lion2.jpg"; 
```
this wont work with a div. only img tags can have src.
please update so the the div wraps around a img element.

AnaCascante replied:
fixed

Jean-Christoffer commented on html/feed/feed.js:
```diff
@@ -1,5 +1,369 @@
+    userImg.src = element.author.avatar ?? "../../img/lion2.jpg"; 
+    userImg.className = 'rounded-circle img-fluid'; 
+    userImg.style = 'width: 50px; height: 50px; display: block;'; 
+     
```
append a img element inside the div

AnaCascante replied:
the append is in line 94.

AnaCascante committed:
fixed some issues

still some issues to solve

Jean-Christoffer commented on html/feed/feed.js:
```diff
@@ -1,5 +1,369 @@
+// likeBtn eventListener 
+
+btnLike.addEventListener ("click", (e)=>{
+  reactToPosts (baseUrl, element.id, token, spanlike)
```
spanlike is typed wrong which causes an error in the console.

AnaCascante replied:
changed it-  spanLike

Jean-Christoffer commented on html/feed/feed.js:
```diff
@@ -1,5 +1,369 @@
+btnComment.setAttribute ('data-bs-target','#commentCollapse${element.id}'); 
+btnComment.setAttribute ('aria-expanded','false'); 
+btnComment.setAttribute ('aria-controls','commentCollapse${element.id}'); 
+    
```
please use correct quotations when using javascript variables inside a "" you need to use correct backtics or else the variable will only be read as a string, it wont open the ability to comment because of this.

AnaCascante replied:
change quotations. replaced with backticks in the necessary places.

Jean-Christoffer commented on html/feed/feed.html:
```diff
@@ -6,36 +6,148 @@
+            <option value="latest">Latest</option>
+            <option value="oldest">Oldest</option>
+             Add more sort options as needed -->
+             
```
you are commenting out to much of the navigation which breaks the layout only the sort tab needs removal not the main tab.

AnaCascante replied:
I reorganized comment out- to let the <div>- tab-content. Then what its comment out now is just the sort div. this one is the one we do not need. if u agree I remove commented sort content

Jean-Christoffer commented on html/feed/feed.html:
```diff
@@ -6,36 +6,148 @@
+            <option value="oldest">Oldest</option>
+             Add more sort options as needed -->
+             
+        </select>
```
a empty select tag here ?

AnaCascante replied:
not empty, part of the sort div - commentet out now.

Jean-Christoffer commented on html/feed/feed.html:
```diff
@@ -6,36 +6,148 @@
+             Add more sort options as needed -->
+             
+        </select>
+        </div>
```
half a div here

AnaCascante replied:
this one is a part of the sort <div> too, its commented out now.

AnaCascante committed:
fixed backticks

AnaCascante committed:
fixed issues

created a post and it is not shown in the feed.html... need to figure out the issue here.

Jean-Christoffer reviewed (changes requested):
very good, all comments are fixed except the append(card) part, after its fixed you can squash and merge : )

AnaCascante committed:
fixed errors

AnaCascante committed:
removed console.log
